Add Breast Cancer Prediction Model

murtaza-sadri-19 opened the pull request:
I have a updated the structure using new repository because the commits were conflicting with each other.
@yashasvini121 kindly let me know about any further changes.

murtaza-sadri-19 commented:
Hi @yashasvini121,
I am sorry that this mess happened but I managed to adhere to Project Structure. Please review this and if any changes please let me know

yashasvini121 reviewed (commented):
Everything looks great so far! However, the page for this model hasn't been created yet. Refer the tutorial on how to create the page. Additionally, make sure to update the `app.py` file accordingly to list your model page. You can ignore model_details for now.

murtaza-sadri-19 commented:
Hi @yashasvini121, I have updated the app.py file.

And I'm confused that is the PR merged or approved. Please help me regarding this.

yashasvini121 commented:
@murtaza-sadri-19, your earlier changes were correct, so those changes were been approved, but not merged.

As long as the PR shows as open, it has not been merged.

murtaza-sadri-19 commented:
> @murtaza-sadri-19, your earlier changes were correct, so those changes were approved but not merged.
> 
> As long as the PR shows as open, it has not been merged.
> 
> P.S. The `app.py` and `pages.json` files have not been updated yet, hope you are working on that.

Got it @yashasvini121, Thanks for helping.
And I have updated App.py, pages and form_configs for Breast Cancer Detection.

murtaza-sadri-19 commented:
Hi @yashasvini121, can you review the changes I have made under App.py, pages and form_configs for Breast Cancer Detection.

yashasvini121 commented on pages/Breast_cancer_Detector.py:
```diff
@@ -0,0 +1,4 @@
+from page_handler import PageHandler
+
+page_handler = PageHandler("pages/pages.json")
+page_handler.render_page("Breast Tumor Classifier Form")
```
Page not found, please update `pages/pages.json`.

yashasvini121 reviewed (changes requested):
You need to add your page details to `pages/pages.json` for it to be rendered by the `page_handler` class.

yashasvini121 commented on models/Breast_Cancer_Detector/predict.py:
`predict.py` and `model.py` have similar codes, either update them, i.e. make them depend on one another, or remove the `model.py` file. and update the code accordingly i.e. fix `pages.json`.

If it gets complicated then leave the file as it is.

yashasvini121 commented on models/Breast_Cancer_Detector/predict.py:
```diff
@@ -0,0 +1,16 @@
+import pickle
+import numpy as np
+
+Loaded_model = pickle.load(open("D:\DS_March\Projects\Breast_cancer_model.pkl", 'rb'))
```
Write relative file path over here

yashasvini121 commented on models/Breast_Cancer_Detector/notebooks/Breast_cancer_Prediction_Ensemble_Hyperparameter.ipynb:
Add a `model_details()` function over here.
The predict.py file should include a properly implemented model_details() function, or the notebook should contain this function to print a detailed model report. The model will not be accepted without this function, as it is essential for generating the necessary model details.

Sorry if I forgot to mention this earlier. Let me know if you have any confusion regarding the `model_details()`

murtaza-sadri-19 commented:
Hi @yashasvini121 , I have modified predict.py accordingly to differ from model.py and added model details function into predict.py which could be called from pages.json and resolved the issue in same.

abhisek2004 commented:
Hi admin @yashasvini121 give me access i want to contribute to this project i will add Model Details in every page and other details. Can i add my own heart failure predction in this project . please give me acess......

murtaza-sadri-19 commented:
@yashasvini121, please review the changes

murtaza-sadri-19 commented:
@yashasvini121, I am working on changes and would commit it soon.

murtaza-sadri-19 commented on models/Breast_Cancer_Detector/predict.py:
```diff
@@ -0,0 +1,83 @@
+    worst_symmetry = float(input("Enter worst symmetry: "))
+    worst_fractal_dimension = float(input("Enter worst fractal dimension: "))
+
+    input_data = [mean_radius, mean_texture, mean_perimeter, mean_area, mean_smoothness, mean_compactness, mean_concavity, mean_concave_points, mean_symmetry, mean_fractal_dimension, radius_error, texture_error, perimeter_error, area_error, smoothness_error, compactness_error, concavity_error, concave_points_error, symmetry_error, fractal_dimension_error, worst_radius, worst_texture, worst_perime ...
```
The actual parameter which contribute to Cancer Detector are less so I am passing only required features to model.

murtaza-sadri-19 commented:
@yashasvini121, Changes made in PR #129

abhisek2004 commented:
@yashasvini121  Admin Can I add Heart Failure Predictive Project

murtaza-sadri-19 commented:
> @yashasvini121 Admin Can I add Heart Failure Predictive Project
@abhisek2004, you could open a new issue regarding that.

abhisek2004 commented:
ok
